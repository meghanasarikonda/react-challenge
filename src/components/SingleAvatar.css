.avatar {
  height: 60px;
  width: 60px;
  border-radius: 50%;
}

.avatar:hover {
  border: 1px solid rgb(155, 160, 163);
  cursor: pointer;
}