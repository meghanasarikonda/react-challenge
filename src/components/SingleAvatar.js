import React, { Component } from 'react';
import './SingleAvatar.css';

class SingleAvatar extends Component {
  constructor(props) {
    super(props);

    this.state = {
      image: this.props.image,
      initial: true
    }

    this.handleClick = (e) => {
      e.preventDefault();
      this.props.openPopover();
    }
  }


  render() {
    console.log('props', this.props.image)

    return (
      <div>
        <img className="avatar"
             src={this.props.image["src"]}
             key={this.props.image["id"]}
             alt={this.props.image["label"]}
             onClick={this.handleClick}
        />
      </div>
    );
  }
}

export default SingleAvatar;