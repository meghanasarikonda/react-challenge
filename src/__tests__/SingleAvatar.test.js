import React from 'react';
import ReactDOM from 'react-dom';
//import Main from './components/Main';
import SingleAvatar from '../components/SingleAvatar';
import Enzyme, { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';


function setup() {
  const props = {
    image: {
      src: 'https://i2.wp.com/beebom.com/wp-content/uploads/2016/01/Reverse-Image-Search-Engines-Apps-And-Its-Uses-2016.jpg',
      id: 1,
      label: 'picture'
    },
    openPopover: jest.fn()
  }
 
  const enzymeWrapper = mount(<SingleAvatar {...props} />)
 
  return {
    props,
    enzymeWrapper
  }
}

describe('SingleAvatar', () => {

  it('should render correctly', () => {
    const {enzymeWrapper} = setup()
    expect(shallowToJson(enzymeWrapper)).toMatchSnapshot();
  });

  it('should have img tag', () => {
    const { enzymeWrapper } = setup()
    expect(enzymeWrapper.find('img').length).toBe(1);
  });

  it('img should have a class named avatar', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find('img').hasClass('avatar')).toBe(true);
  });

  it('checking component props', () => {
    const { enzymeWrapper } = setup();
    const SingleAvatarProps = enzymeWrapper.find('SingleAvatar').props();

    expect(Object.keys(SingleAvatarProps).length).toBe(2);
    expect(typeof SingleAvatarProps.image).toBe('object');
    expect(typeof SingleAvatarProps.openPopover).toBe('function');
  });

  it('should have handleClick method', () => {
    const {enzymeWrapper} = setup();
    expect(typeof enzymeWrapper.instance().handleClick).toBe('function');
  });

  it('click event', () => {
    window.alert = jest.fn();
    const { enzymeWrapper } = setup();
    const SingleAvatarProps = enzymeWrapper.find('SingleAvatar').props();

    enzymeWrapper.find('img').simulate('click');
    expect(enzymeWrapper.props().openPopover).toHaveBeenCalled();
  });

})