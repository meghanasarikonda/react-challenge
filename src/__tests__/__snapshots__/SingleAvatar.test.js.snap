// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleAvatar should render correctly 1`] = `
<SingleAvatar
  image={
    Object {
      "id": 1,
      "label": "picture",
      "src": "https://i2.wp.com/beebom.com/wp-content/uploads/2016/01/Reverse-Image-Search-Engines-Apps-And-Its-Uses-2016.jpg",
    }
  }
  openPopover={[Function]}
>
  <div>
    <img
      alt="picture"
      className="avatar"
      key="1"
      onClick={[Function]}
      src="https://i2.wp.com/beebom.com/wp-content/uploads/2016/01/Reverse-Image-Search-Engines-Apps-And-Its-Uses-2016.jpg"
    />
  </div>
</SingleAvatar>
`;
